eclair-core/src/test/scala/fr/acinq/eclair/payment/Bolt12InvoiceSpec.scala

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2022 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.payment
+
+import fr.acinq.bitcoin.Crypto.{PrivateKey, PublicKey}
+import fr.acinq.bitcoin.{ByteVector32, Crypto}
+import fr.acinq.eclair.crypto.Sphinx.RouteBlinding
+import fr.acinq.eclair.payment.Bolt12Invoice._
+import fr.acinq.eclair.wire.protocol.OfferCodecs.invoiceTlvCodec
+import fr.acinq.eclair.wire.protocol.Offers._
+import fr.acinq.eclair.wire.protocol.TlvStream
+import fr.acinq.eclair.{CltvExpiryDelta, Features, MilliSatoshi, MilliSatoshiLong, TimestampSecond, randomBytes, randomBytes32, randomKey}
+import org.scalatest.funsuite.AnyFunSuite
+import scodec.bits._
+
+class Bolt12InvoiceSpec extends AnyFunSuite {
```

Comment:
We really need a lot more unit tests on this and on offers, given the sheer number of codecs and types that we're adding to the codebase.

A good rule of thumb is to try to find everything that could go wrong and write a unit test for it.